bundles/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/services/DeviceManager.java

```diff
@@ -115,48 +117,63 @@ private void searchDevice() {
         try {
             final InetAddress multicastAddress = InetAddress.getByName(MULTI_CAST_HOST);
 
-            final List<NetworkInterface> networkInterfaces = Collections.list(NetworkInterface.getNetworkInterfaces());
+            final List<NetworkInterface> networkInterfaces = Collections.list(NetworkInterface.getNetworkInterfaces())
+                    .stream().filter(device -> {
+                        try {
+                            return device.isUp() && !device.isLoopback();
+                        } catch (SocketException e) {
+                            logger.debug("Failed to get device information", e);
+                            return false;
+                        }
+                    }).collect(Collectors.toList());
 
             executorService = Executors.newFixedThreadPool(networkInterfaces.size());
```

Comment:
There were no changes on this line. But it looks like it creates a  new fixed thread pool each time search is started :cry:

Reply:
It would also be better to fetch the network devices  only once. 

I didn't know if I should keep my changes minimal or mix the bugfix and a refactoring.

Reply:
In general I don't think it matters much. Only if the refactoring is large it could be the review process will become longer. So if you have a small fix and want that merged then keeping the change small increases the chance to get it merged quicker.